src/balances/balances.service.ts

```diff
@@ -1,16 +1,89 @@
 import { Injectable } from '@nestjs/common';
-import { Balance } from '../services/transaction-service/entities/balance.entity';
+import { Balance as TransactionServiceBalance } from '../services/transaction-service/entities/balance.entity';
+import { ExchangeService } from '../services/exchange/exchange.service';
+import { TokenInfo } from '../common/entities/tokeninfo.entity';
+import { TokenType } from '../common/entities/tokentype.entity';
+import { Balances } from './entities/balances.entity';
+import { ConfigService } from '../services/config-service/config-service.service';
 import { TransactionServiceManager } from '../services/transaction-service/transaction-service.manager';
+import { NativeCurrency } from '../services/config-service/entities/native.currency.entity';
+import { Balance } from './entities/balance.entity';
 
 @Injectable()
 export class BalancesService {
   constructor(
-    private readonly transactionServiceManager: TransactionServiceManager,
+    private readonly safeConfigService: ConfigService,
+    private readonly safeTransactionManager: TransactionServiceManager,
+    private readonly exchangeService: ExchangeService,
   ) {}
 
-  async getBalances(chainId: string, safeAddress: string): Promise<Balance[]> {
-    const transactionService =
-      await this.transactionServiceManager.getTransactionService(chainId);
-    return transactionService.getBalances(safeAddress);
+  async getBalances(
+    chainId: string,
+    safeAddress: string,
+    fiatCode: string,
+  ): Promise<Balances> {
+    const safeTransactionService =
+      await this.safeTransactionManager.getTransactionService(chainId);
+    const txServiceBalances: TransactionServiceBalance[] =
+      await safeTransactionService.getBalances(safeAddress);
+
+    const usdToFiatRate: number = await this.exchangeService.convertRates(
+      fiatCode,
+      'USD',
```

Comment:
I would extract this string into a constant somewhere on top of this file.

Reply:
Extracted here: [e3deb67](https://github.com/5afe/safe-client-gateway-nest/pull/29/commits/e3deb6736c1038922b1ba19d8aa892c305eb6bcf)